pkg/kv/kvserver/gc/gc.go

```diff
@@ -46,6 +48,19 @@ const (
 	// Raft command for each GCRequest does not significantly exceed
 	// this threshold.
 	KeyVersionChunkBytes = base.ChunkRaftCommandThresholdBytes
+	// defaultClearRangeMinKeys is min number of versions we will try to
+	// remove using clear range. If we have less than this number, using pebble
+	// range tombstone is more expensive than using multiple point deletions.
+	defaultClearRangeMinKeys = 10000
```

Comment:
Yeah those numbers are way off that ideal and very conservative. I actually run a benchmark to see if I have a 1M keys and then delete series of keys leaving a single remaining key in between. Then compare how it would perform with all point clear vs range clear. I also added third one where clear range would be nested progressively wider to mimic how multiple versions behave if you keep deleting longer and longer history up to next key if pebble range tombstones are not compacted. For iterating all remaining keys, benefits are appearing at 100 keys already. Starting at 2x improvement on iteration and 30% at 100 keys which is lowered to 70% improvement and 20% if there are two nested range tombstones. At 1000 improvement is 3x for scans and 2.5 for seeks.
My concern is that if we want to enable them on low number of keys then we must batch multiple of those together. Otherwise we will run too many individual requests. Let's say we have 100 byte keys, then we can delete 2.5K keys with point deletions per batch, so we need to be in the same ballpark with number of clear range request counts.
Maybe we could lower that to 2000.

Reply:
Yeah, something like 2000 seems reasonable.